azext_iot/monitor/handlers/central_handler.py

```diff
@@ -0,0 +1,120 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import List
+
+from azext_iot.central.providers.device_provider import CentralDeviceProvider
+from azext_iot.monitor.utility import stop_monitor, get_loop
+from azext_iot.monitor.handlers.common_handler import CommonHandler
+from azext_iot.monitor.parsers.central_parser import CentralParser
+from azext_iot.monitor.parsers.issue import Severity, Issue
+
+
+class CentralHandler(CommonHandler):
+    """
+    Handles messages as they are read from egress event hub.
+
+    Keyword Args:
+        timeout             (int)       Maximum duration to keep listening for events. Use 0 "forever". Defaults to 0.
+        max_messages        (int)       Maximum number of messages to read. Use 0 "forever". Defaults to 0.
+        progress_interval   (int)       number of messages to wait between printing progress. Defaults to 5.
+        minimum_severity    (Severity)  minimum severity for reporting issues. Defaults to Severity.warning.
+        device_id           (str)       only process messages sent by this device
+        properties          (list)      list of properties to extract from message headers
+        output              (str)       output format (json, yaml, etc)
+    """
+
+    def __init__(self, central_device_provider: CentralDeviceProvider, **kwargs):
+        super(CentralHandler, self).__init__(**kwargs)
+        self.central_device_provider = central_device_provider
+
+        self.timeout = kwargs.get("timeout") or 0
+        self.max_messages = kwargs.get("max_messages") or 0
+        self.progress_interval = kwargs.get("progress_interval") or 5
+        self.minimum_severity = kwargs.get("minimum_severity") or Severity.warning
+
+        self.messages = []
+        self.issues: List[Issue] = []
+
+        if self.timeout:
+            loop = get_loop()
+            # the monitor takes a few seconds to start
+            loop.call_later(self.timeout + 5, self._quit_on_timeout)
```

Comment:
This timeout is different from the currently released timeout. The original intent for timeout is to indicate how long to maintain a connection without receiving message and its handled at the amqp client level. Versus, here where its a static timeout and close connection. Is that intended?

Reply:
Yes, our monitor has slightly different needs. But after chatting with the team earlier today we decided to move it to another variable so that the experience of using --timeout is consistent with the other monitors